### PR TITLE
Modify application api endpoints to connect to database

### PR DESCRIPTION
**Pull request by PriscillaSam**
#### What does this PR do?
Modify api endpoints to connect to a database.

#### Description of Task to be completed?
Have the following api endpoints working with a database.
**POST** `api/v1/auth/signup` - Register user.
**POST** `api/v1/auth/login` - Login user.

**GET**  `api/v1/questions` - Get all questions.
**GET**  `api/v1/questions/:id` - Get a particular question by id.
**GET** `api/v1/users/questions` - Get all questions for a particular user.
**POST** `api/v1/questions` - Post a question.
**DELETE**  `api/v1/questions/:id` - Delete a question.

**POST** `api/v1/questions/:id/answers` - Post an answer.
**POST** `api/v1/questions/:questionId/answers/:answerId` - Mark an answer as accepted.
**POST** `api/v1/answers/:id` - Vote an answer.

#### How should this be manually tested?
* Clone the repo using `git clone https://github.com/PriscillaSam/Stackoverflow-Lite.git`.
* run `git checkout challenge3`.
* run `npm install` and `npm start`.
* Open up Postman and test the endpoints above.

#### Any background context you want to provide?
None.

#### What are the relevant pivotal tracker stories?
#159962365, #159960444, #160021000, #159965343, #160038467, #160021517, #160021682, #160022015, #160021774, #160022206, #160101059

### CODE REVIEW

File: server/config/db.config.js [hunk 1]
```diff
@@ -1,12 +1,19 @@
 import pg from 'pg';
 import dotenv from 'dotenv';
+import parseUrl from 'parse-database-url';
+
 import config from './config';
 
 dotenv.config();
 const { Pool } = pg;
+let configInfo;
```
Comment (Kenec) [inline server/config/db.config.js:9]: What if the NODE_ENV is test? configInfo will be undefined variable. You may want to take care of a situation where the environment is not known.

File: server/controllers/answer.js [hunk 2]
```diff
@@ -1,152 +1,221 @@
...
+        client.query(questionQueries.getQuestion(questionId))
+          .then((response) => {
+            const [questionObj] = response.rows;
+            if (!questionObj) {
+              return errors.notFound(res, 'question');
+            }
+            client.query(answerQueries.getAnswersByQId(questionId))
+              .then((answerResponse) => {
+                const answers = answerResponse.rows;
+                if (answers.find(ans => ans.userid === userId)) {
```
Comment (Kenec) [inline server/controllers/answer.js]: Your variable name should be descriptive. ans can simple be named answer.

File: server/controllers/answer.js [hunk 3]
```diff
@@ -1,152 +1,221 @@
...
+            if (!questionObj) {
+              return errors.notFound(res, 'question');
+            }
+            client.query(answerQueries.getAnswer(answerId))
+              .then((ansResponse) => {
+                const [existingAnswer] = ansResponse.rows;
+                if (!existingAnswer) {
+                  return errors.notFound(res, 'answer');
+                }
+                if (existingAnswer.questionid !== questionId) {
```
Comment (Kenec) [inline server/controllers/answer.js:74]: - variable should be named in camelCas. questionid should be questionId
- 404 status code is misused here. If the answer exists (then definitely not 404) but belongs to another question, then return appropriate status code to depict this kind of conflict
_

File: server/controllers/answer.js [hunk 4]
```diff
@@ -1,152 +1,221 @@
...
+                      .updateAnswer(answerId, 'answer', answer))
+                      .then((updateRes) => {
+                        const [updatedAnswer] = updateRes.rows;
+                        return res.status(200).json({
+                          status: 'success',
+                          message: 'you have updated your answer',
+                          updatedAnswer,
+                        });
+                      });
+                  } else if (questionObj.userid !== userId) {
```
Comment (Kenec) [inline server/controllers/answer.js:97]: - This kind of checks should be abstracted to a middleware. We do not want our controller to be doing a task on unchecked parameters
-

**Review by Kenec — CHANGES_REQUESTED** [review]

File: server/controllers/answer.js (hunk 3 above)
Comment (PriscillaSam) [reply server/controllers/answer.js:74]: The variable name is that way here because sql somehow modifies camel casing to all lowercase

**Pull request merged**